api/provider/aws/test_aws/load_balancer_create_test.go

```diff
@@ -0,0 +1,275 @@
+	createLoadBalancerInput.SetListeners(listeners)
+
+	validateFN := func(input *elb.CreateLoadBalancerInput) {
+		assert.NotNil(t, input.Listeners)
+		assert.NotNil(t, input.LoadBalancerName)
+
+		for _, listener := range input.Listeners {
+			err := listener.Validate()
+			assert.NoError(t, err)
+		}
+	}
+
+	mockAWS.ELB.EXPECT().
+		CreateLoadBalancer(createLoadBalancerInput).
+		Do(validateFN).
+		Return(&elb.CreateLoadBalancerOutput{}, nil)
+
+	healthCheck := &elb.HealthCheck{}
+	healthCheck.SetTarget("HTTPS:443/path/to/site")
+	healthCheck.SetInterval(int64(20))
+	healthCheck.SetTimeout(int64(15))
+	healthCheck.SetHealthyThreshold(int64(4))
+	healthCheck.SetUnhealthyThreshold(int64(3))
+
+	configureHealthCheckInput := &elb.ConfigureHealthCheckInput{}
+	configureHealthCheckInput.SetLoadBalancerName("l0-test-lb_id")
+	configureHealthCheckInput.SetHealthCheck(healthCheck)
+
+	mockAWS.ELB.EXPECT().
+		ConfigureHealthCheck(configureHealthCheckInput).
+		Return(&elb.ConfigureHealthCheckOutput{}, nil)
+
+	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Create(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "lb_id", result)
+
+	expectedTags := models.Tags{
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name",
+		},
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range expectedTags {
+		assert.Contains(t, tagStore.Tags(), tag)
+	}
+}
+
+func TestLoadBalancerCreateDefaults(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+	mockConfig.EXPECT().VPC().Return("vpc_id").AnyTimes()
+	mockConfig.EXPECT().Region().Return("region").AnyTimes()
+	mockConfig.EXPECT().AccountID().Return("123456789012")
+	mockConfig.EXPECT().PrivateSubnets().Return([]string{"priv1", "priv2"}).AnyTimes()
+
+	defer provider.SetEntityIDGenerator("lb_id")()
+
+	req := models.CreateLoadBalancerRequest{
+		LoadBalancerName: "lb_name",
+		EnvironmentID:    "env_id",
+		Ports:            []models.Port{},
+		HealthCheck:      models.HealthCheck{},
+	}
+
+	readSGHelper(mockAWS, "l0-test-env_id-env", "env_sg")
+	createSGHelper(t, mockAWS, "l0-test-lb_id-lb", "vpc_id")
+	readSGHelper(mockAWS, "l0-test-lb_id-lb", "lb_sg")
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(gomock.Any()).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	mockAWS.IAM.EXPECT().
+		CreateRole(gomock.Any()).
+		Return(&iam.CreateRoleOutput{}, nil)
+
+	mockAWS.IAM.EXPECT().
+		PutRolePolicy(gomock.Any()).
+		Return(&iam.PutRolePolicyOutput{}, nil)
+
+	listeners := []*elb.Listener{listenerHelper(nil)}
```

Comment:
Going off of my other comment - we should definitely call something like `listenerHelper(defaults.DefaultLBPort`) instead of using `nil`